docs/modules/components/pages/processors/ollama_chat.adoc

```diff
@@ -123,6 +133,170 @@ This field supports xref:configuration:interpolation.adoc#bloblang-queries[inter
+
+
+*Type*: `int`
+
+
+=== `num_keep`
+
+Specify the number of tokens from the initial prompt to retain when the model resets its internal context. By default, this value is set to 4. Use -1 to retain all tokens from the initial prompt.
+
+
+*Type*: `int`
+
+
+=== `seed`
+
+Sets the random number seed to use for generation. Setting this to a specific number will make the model generate the same text for the same prompt.
+
+
+*Type*: `int`
+
+
+=== `top_k`
+
+Reduces the probability of generating nonsense. A higher value (e.g. 100) will give more diverse answers, while a lower value (e.g. 10) will be more conservative.
+
+
+*Type*: `int`
+
+
+=== `top_p`
+
+Works together with top-k. A higher value (e.g., 0.95) will lead to more diverse text, while a lower value (e.g., 0.5) will generate more focused and conservative text.
+
+
+*Type*: `float`
+
+
+=== `repeat_penalty`
+
+Sets how strongly to penalize repetitions. A higher value (e.g., 1.5) will penalize repetitions more strongly, while a lower value (e.g., 0.9) will be more lenient.
+
+
+*Type*: `float`
+
+
+=== `presence_penalty`
+
+Positive values penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics.
+
+
+*Type*: `float`
+
+
+=== `frequency_penalty`
+
+Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim.
+
+
+*Type*: `float`
+
+
+=== `stop`
+
+Sets the stop sequences to use. When this pattern is encountered the LLM will stop generating text and return.
+
+
+*Type*: `array`
+
+
+=== `runner`
+
+Options for the model runner that are used when the model is first loaded into memory.
+
+
+*Type*: `object`
+
+
+=== `runner.context_size`
+
+Sets the size of the context window used to generate the next token.
+
+
+*Type*: `int`
+
+
+=== `runner.batch_size`
+
+The maximum number of requests to process in parallel.
+
+
+*Type*: `int`
+
+
+=== `runner.gpu_layers`
+
+This option allows offloading some layers to the GPU for computation. Generally results in increased performance. By default the runtime decides the number of layers dynamically.
+
+
+*Type*: `int`
+
+
+=== `runner.threads`
+
+Set the number of threads to use during generation. For optimal performance, it is recommended to set this value to the number of physical CPU cores your system has. By default the runtime decides the optimal number of threads.
+
+
+*Type*: `int`
+
+
+=== `runner.use_mmap`
+
+If supported, map the model into memory. This allows the system to load only the necessary parts of the model as needed.
+
+
+*Type*: `bool`
+
+
+=== `runner.use_mlock`
+
+Lock the model in memory, preventing it from being swapped out when memory-mapped. This can improve performance but trades away some of the advantages of memory-mapping by requiring more RAM to run and potentially slowing down load times as the model loads into RAM.
```

Comment:
```suggestion
Lock the model in memory, preventing it from being swapped out when memory-mapped. This option can improve performance but reduces some of the advantages of memory-mapping because it uses more RAM to run and can slow down load times as the model loads into RAM.
```